flocker/control/_persistence.py

```diff
@@ -87,7 +93,58 @@ def decode_object(dictionary):
                        [_DEPLOYMENT_FIELD], [])
 
 
-class ConfigurationPersistenceService(Service):
+class LeaseService(Service):
+    """
+    Manage leases.
+    In particular, clear out expired leases once a second.
+
+    :ivar _reactor: A ``twisted.internet.reactor`` implementation.
+    :ivar _persistence_service: The persistence service to act with.
+    :ivar _lc: A ``twisted.internet.task.LoopingCall`` run every second
+        to update the configured leases by releasing leases that have
+        expired.
+    """
+    def __init__(self, reactor, persistence_service):
+        if reactor is None:
+            reactor = default_reactor
+        self._reactor = reactor
+        self._persistence_service = persistence_service
+
+    def startService(self):
+        self._lc = LoopingCall(self._expire)
+        self._lc.clock = self._reactor
+        self._lc.start(1)
+
+    def stopService(self):
+        self._lc.stop()
+
+    def _expire(self):
+        now = datetime.now(tz=UTC)
+        return update_leases(lambda leases: leases.expire(now),
+                             self._persistence_service)
+
+
+def update_leases(transform, persistence_service):
+    """
+    Update the leases configuration in the persistence service.
+
+    :param transform: A function to execute on the currently configured
+        leases to manipulate their state.
+    :param persistence_service: The persistence service to which the
+        updated configuration will be saved.
+
+    :return Deferred: Fires when the persistence service has saved.
+    """
+    # XXX we cannot manipulate leases in this branch since the configuration
+    # doesn't know anything about them yet. See FLOC-2731 subtasks.
+    # configuration = persistence_service.get()
+    # return persistence_service.save(
+    #     configuration.set("leases", transform(configuration.leases)))
```

Comment:
Please delete this commented out code for now. If you need it for reference, then put it in the appropriate FLOC-2731 subtask.